lib/server/controllers/qcFlag.controller.js

```diff
@@ -82,7 +82,61 @@ const listQcFlagsHandler = async (req, res) => {
     }
 };
 
+// eslint-disable-next-line valid-jsdoc
+/**
+ * Create QcFlag
+ */
+const createQcFlagHandler = async (req, res) => {
+    const validatedDTO = await dtoValidator(
+        DtoFactory.bodyOnly(Joi.object({
+            from: Joi.number().required().allow(null),
+            to: Joi.number().required().allow(null),
+            comment: Joi.string().optional().allow(null),
+            flagTypeId: Joi.number().required(),
+            runNumber: Joi.number().required(),
+            dplDetectorId: Joi.number().required(),
+            dataPassId: Joi.number(),
+            simulationPassId: Joi.number(),
+        }).xor('dataPassId', 'simulationPassId')),
+        req,
+        res,
+    );
+    if (validatedDTO) {
+        try {
+            const {
+                from,
+                to,
+                comment,
+                flagTypeId,
+                runNumber,
+                dplDetectorId,
+                dataPassId,
+                simulationPassId,
+            } = validatedDTO.body;
+            const parameters = { from, to, comment };
+            const relations = {
+                user: { externalUserId: validatedDTO.session.externalId },
+                flagTypeId,
+                runNumber,
+                dplDetectorId,
+                dataPassId,
+                simulationPassId,
+            };
+            let createdFlag;
+            if (relations.dataPassId) {
```

Comment:
Why not using directly dataPassId ?

Reply:
Neither dataPassId not simulationPassId is property of QC flag, there are junction tables for both

Reply:
in the if?
You have `relations.dataPassId = dataPassId` then you use `relations.dataPassId`, I would rather use dataPassId directly

Reply:
ok, I absolutely misunderstood, applied